core-bundle/src/Resources/contao/drivers/DC_Folder.php

```diff
@@ -1977,11 +1978,7 @@ public function source()
 					$objVersions->restore(Input::post('version'));
 
 					// Purge the script cache (see #7005)
-					if ($objFile->extension == 'css' || $objFile->extension == 'scss' || $objFile->extension == 'less')
```

Comment:
The `js` extension was missing here but I don't know why. We might add a fix for Contao 4.4/4.9 if this wasn't intentional and adjust this PR else.

Reply:
According to https://github.com/contao/core/issues/7005, the script cache was only problematic for .css, .scss and .less files, because these files were generated and/or combined. Not sure if we also combine .js files that are added to a page layout as external files? If so, adding `js` is correct.

Reply:
> Not sure if we also combine .js files that are added to a page layout as external files?

External files selected in the page layout do not get combined. But it is also possible to add JS files via *js_* templates using `$GLOBALS['TL_JAVASCRIPT'][] = …` so purging the script cache for `js` files is correct IMO.